elyra/pipeline/pipeline_definition.py

```diff
@@ -457,58 +488,26 @@ def validate(self) -> list:
 
         return validation_issues
 
-    def coerce_kv_pair_properties(self):
+    def propagate_pipeline_default_properties(self):
         """
-        Convert pipeline- and node-level list properties that have been identified
-        as sets of key-value pairs from a plain list type to the KeyValueList list
-        type. Properties are identified as key-value pair properties when they set
-        keyValueEntries to true in the pipeline JSON canvas properties definition.
+        For any default pipeline properties set (e.g. runtime image, volume), propagate
+        the values to any nodes that do not set their own value for that property.
         """
-        all_pipeline_properties = PipelineDefinition.get_canvas_properties_from_template(
-            package_name="templates/pipeline", template_name="pipeline_properties_template.jinja2"
-        )
-        kv_pair_properties = PipelineDefinition.get_key_value_pair_properties(all_pipeline_properties)
-
+        kv_pair_properties = PipelineDefinition.get_pipeline_kv_pair_properties()
         pipeline_default_properties = self.primary_pipeline.get_property(pipeline_constants.PIPELINE_DEFAULTS, {})
-        for pipeline_default_prop, pipeline_default_value in pipeline_default_properties.items():
-            if pipeline_default_prop in kv_pair_properties:
-                # Coerce plain list to KeyValueList
-                self.set_pipeline_default_property(pipeline_default_prop, KeyValueList(pipeline_default_value))
 
-        generic_node_properties = PipelineDefinition.get_canvas_properties_from_template(
-            package_name="templates/components", template_name="generic_properties_template.jinja2"
-        )
+        # Convert any key/value list pipeline default properties to the KeyValueList type
+        self.primary_pipeline.convert_kv_pair_properties(kv_pair_properties)
```

Comment:
This is a nit, but I think it increases readability/maintainability.  Could we move the initialization of `pipeline_default_properites` directly before the `for` loop since its only used within, then group the other two statements since they're dependent?

```python
        # Convert any key/value list pipeline default properties to the KeyValueList type
        kv_properties = PipelineDefinition.get_kv_properties()
        self.primary_pipeline.convert_kv_properties(kv_properties)

        pipeline_default_properties = self.primary_pipeline.get_property(pipeline_constants.PIPELINE_DEFAULTS, {})
````

Reply:
I had the same thought while I was refactoring, not sure why I didn't pull the trigger on that! 😀